Dockerfile

```diff
@@ -1,17 +1,18 @@
-FROM registry.access.redhat.com/ubi8-minimal as builder
-RUN curl --silent --location https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo
-RUN microdnf -y install yarn nodejs
+FROM registry.access.redhat.com/ubi8/nodejs-12:latest as builder
 COPY . /mig-ui
 WORKDIR /mig-ui
-RUN yarn
-RUN yarn build
+USER root
```

Comment:
Do you need to set root? `microdnf` also needs to be run as `root` and the user was not specified in the previous version.

Reply:
You can't run dnf without root. It wasn't set before because no user is set on the base ubi container so it runs as root.

```
$ docker run --rm -it --entrypoint whoami registry.access.redhat.com/ubi8
root
```

```
$ docker run --rm -it --entrypoint whoami registry.access.redhat.com/ubi8/nodejs-12:latest
default
```

This is a build container that is discarded.

It also does not affect what this or the application container is necessarily run as, just sets a default from here on out (and has zero effect on the runtime container created later).

```
$ docker run --rm -it --entrypoint whoami quay.io/jmontleon/mig-ui:latest
default
```

This is the image I built with the Dockerfile:
```
$ oc get deployment/migration-ui -o yaml | grep -i '\- image:'
      - image: quay.io/jmontleon/mig-ui:latest

$ oc exec -it deployment/migration-ui -- whoami
1000270000
```

Reply:
You can remove the `USER root` line and try for yourself:
```
$ docker build .
Sending build context to Docker daemon  392.9MB
Step 1/14 : FROM registry.access.redhat.com/ubi8/nodejs-12:latest as builder
 ---> 8a961c0b3cbc
Step 2/14 : COPY . /mig-ui
 ---> 45d8b1ea22b1
Step 3/14 : WORKDIR /mig-ui
 ---> Running in d0bed3f0e053
Removing intermediate container d0bed3f0e053
 ---> 54eac7341e60
Step 4/14 : RUN dnf config-manager --add-repo https://dl.yarnpkg.com/rpm/yarn.repo &&     dnf -y install yarn && yarn && yarn build
 ---> Running in e85998215e20
2020-10-12 17:13:04,332 [ERROR] dnf:6:MainThread @logutil.py:194 - [Errno 13] Permission denied: '/var/log/rhsm/rhsm.log' - Further logging output will be written to stderr
Not root, Subscription Management repositories not updated
Error: This command has to be run under the root user.
The command '/bin/sh -c dnf config-manager --add-repo https://dl.yarnpkg.com/rpm/yarn.repo &&     dnf -y install yarn && yarn && yarn build' returned a non-zero code: 1
```

Reply:
Thanks. I had not thought that the ubi8/ubi image wouldn't use root as default user.